app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -337,6 +354,33 @@ public Task<Map<String, Object>> fetchNearbyPins(@NonNull Location location) {
                 .continueWith(task -> (Map<String, Object>) task.getResult().getData());
     }
 
+    // TODO: improve fetch efficiency using query
+    public void fetchSocials() {
+        if (auth.getUid() == null) {
+            throw new IllegalStateException("User must be logged in to fetch followers");
+        }
+        followerIds = new HashSet<>();
+        followingIds = new HashSet<>();
+        db.collection("social").document(auth.getUid()).get().continueWith(task -> {
```

Comment:
This should probably be an `onCompleteListener` rather than a `continueWith` since you're not returning anything from the `continueWith` method.